### PR TITLE
COS integration tests

### PR DESCRIPTION
**Pull request by neoaggelos**
### Summary

Write a proper integration test for cos-lite

### Changes

Replace `test_observability` with `test_cos`. The following functionality is tested:

- Update the charm to use the latest version of all cos libraries (`python src/hack/update_libs.py`)
- Deploy cos-lite on top of the MicroK8s charm.
- Expose cos-lite endpoints via a cross-model relation to the microk8s charm model
- Deploy grafana-agent and relate microk8s charm with COS
- Ensure that Kubernetes metrics can be retrieved from COS

### Notes

Drop Juju 2.9 from our testing, since cos-lite depends on Juju 3+

### CODE REVIEW

File: tests/integration/test_cos.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        try:
+            LOG.info("Consume cos-lite and relate with grafana-agent")
+            await e.model.consume(f"admin/{k8s_model_name}.prometheus", "prometheus")
+            await e.model.consume(f"admin/{k8s_model_name}.loki", "loki")
+            await e.model.consume(f"admin/{k8s_model_name}.grafana", "grafana")
+            await e.model.add_relation("grafana-agent", "prometheus")
+            await e.model.add_relation("grafana-agent", "loki")
+            await e.model.add_relation("grafana-agent", "grafana")
+
+            await e.model.wait_for_idle(["microk8s", "grafana-agent"], timeout=20 * 60)
```
Comment (neoaggelos) [inline tests/integration/test_cos.py:84]: @sed-i here's where we get the timeout issues. pretty much, after relating to cos components, grafana-agent's hooks will check to see if the grafana-agent snap is installed, and that operation has a 5 second timeout (which is a bit low for the github runners).

File: tests/integration/test_cos.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        await e.model.wait_for_idle(["microk8s", "grafana-agent"])
+
+    microk8s_unit: Unit = e.model.applications["microk8s"].units[0]
+    grafana_agent_app: Application = e.model.applications["grafana-agent"]
+
+    # TODO(neoaggelos): this is a terrible workaround but solves timeout issues in GitHub actions
+    await run_unit(
+        microk8s_unit,
+        "sudo sed -i 's,timeout: float = 5.0,timeout: float = 20.0,' /var/lib/juju/agents/unit-grafana-agent*/charm/lib/charms/operator_libs_linux/v1/snap.py",  # noqa
+    )
```
Comment (neoaggelos) [inline tests/integration/test_cos.py:48]: here's the hack @sed-i

File: tests/integration/test_cos.py (hunk 1 above)
Comment (neoaggelos) [reply tests/integration/test_cos.py:84]: https://github.com/canonical/grafana-agent-k8s-operator/pull/247 might help with this

**Review by ktsakalozos — APPROVED** [review]
LGTM +1

**Pull request merged**